integration-tests/src/main/java/org/apache/polaris/service/it/test/CatalogFederationIntegrationTest.java

```diff
@@ -0,0 +1,239 @@
+    newUserCredentials = managementApi.createPrincipal(PRINCIPAL_NAME);
+
+    FileStorageConfigInfo storageConfig =
+        FileStorageConfigInfo.builder()
+            .setStorageType(StorageConfigInfo.StorageTypeEnum.FILE)
+            .setAllowedLocations(List.of(baseLocation.toString()))
+            .build();
+
+    CatalogProperties catalogProperties = new CatalogProperties(baseLocation.toString());
+
+    Catalog localCatalog =
+        PolarisCatalog.builder()
+            .setType(Catalog.TypeEnum.INTERNAL)
+            .setName(LOCAL_CATALOG_NAME)
+            .setProperties(catalogProperties)
+            .setStorageConfigInfo(storageConfig)
+            .build();
+    managementApi.createCatalog(localCatalog);
+
+    CatalogGrant catalogGrant =
+        CatalogGrant.builder()
+            .setType(CatalogGrant.TypeEnum.CATALOG)
+            .setPrivilege(CatalogPrivilege.TABLE_WRITE_DATA)
+            .build();
+    managementApi.addGrant(LOCAL_CATALOG_NAME, CATALOG_ROLE_NAME, catalogGrant);
+    managementApi.assignPrincipalRole(PRINCIPAL_NAME, PRINCIPAL_ROLE_NAME);
+    CatalogRole localCatalogAdminRole =
+        managementApi.getCatalogRole(LOCAL_CATALOG_NAME, CATALOG_ROLE_NAME);
+    managementApi.grantCatalogRoleToPrincipalRole(
+        PRINCIPAL_ROLE_NAME, LOCAL_CATALOG_NAME, localCatalogAdminRole);
+
+    AuthenticationParameters authParams =
+        OAuthClientCredentialsParameters.builder()
+            .setAuthenticationType(AuthenticationParameters.AuthenticationTypeEnum.OAUTH)
+            .setTokenUri(endpoints.catalogApiEndpoint().toString() + "/v1/oauth/tokens")
+            .setClientId(newUserCredentials.getCredentials().getClientId())
+            .setClientSecret(newUserCredentials.getCredentials().getClientSecret())
+            .setScopes(List.of("PRINCIPAL_ROLE:ALL"))
+            .build();
+    ConnectionConfigInfo connectionConfig =
+        IcebergRestConnectionConfigInfo.builder()
+            .setConnectionType(ConnectionConfigInfo.ConnectionTypeEnum.ICEBERG_REST)
+            .setUri(endpoints.catalogApiEndpoint().toString())
+            .setRemoteCatalogName(LOCAL_CATALOG_NAME)
+            .setAuthenticationParameters(authParams)
+            .build();
+    ExternalCatalog externalCatalog =
+        ExternalCatalog.builder()
+            .setType(Catalog.TypeEnum.EXTERNAL)
+            .setName(EXTERNAL_CATALOG_NAME)
+            .setConnectionConfigInfo(connectionConfig)
+            .setProperties(catalogProperties)
+            .setStorageConfigInfo(storageConfig)
+            .build();
+    managementApi.createCatalog(externalCatalog);
+
+    managementApi.addGrant(EXTERNAL_CATALOG_NAME, CATALOG_ROLE_NAME, catalogGrant);
+    CatalogRole externalCatalogAdminRole =
+        managementApi.getCatalogRole(EXTERNAL_CATALOG_NAME, CATALOG_ROLE_NAME);
+    managementApi.grantCatalogRoleToPrincipalRole(
+        PRINCIPAL_ROLE_NAME, EXTERNAL_CATALOG_NAME, externalCatalogAdminRole);
```

Comment:
We only have one test right now so it's fine, but if we add more this should get refactored out